### PR TITLE
Feature/cars endpoints

### PR DESCRIPTION
**Pull request by PVARGASM1**
## Summary

- Added new CRUD  about Cars.
- Changes were tested in local env.

### CODE REVIEW

**Review by danielhincapievargas — APPROVED** [review]

File: src/api/cars/cars.controller.ts [hunk 1]
```diff
@@ -1,20 +1,126 @@
...
+      air_conditioner: car.air_conditioner,
+      transmition: car.transmition,
+      fare_km: car.fare_km,
+      driver_id: car.driver_id,
+      created_at: car.created_at,
+      updated_at: car.updated_at
+    }
 
-  return res.status(200).json(cars)
+    return res.json({message: "Car have been found successfully", car});
```
Comment (ht1204) [inline src/api/cars/cars.controller.ts]: I consider return is not necessary for sending a response, just send the response and remember add it an status, i.e. something like:
`res.status(201).json({message: "Car have been found successfully", car});`

File: src/api/cars/cars.controller.ts [hunk 2]
```diff
@@ -1,20 +1,126 @@
...
-  const data = req.body
 
-  const carCreated = await createCar(data)
+export const createCarHandler = async (req: AuthRequest, res: Response) => {
+  try {
+    const data = req.body
+
+    const carCreated = await createCar(data)
+
+    return res.status(201).json({message: "Car was created successfully", carCreated})
```
Comment (ht1204) [inline src/api/cars/cars.controller.ts]: return could not be necessary, just res.

File: src/api/cars/cars.service.ts [hunk 3]
```diff
@@ -4,30 +4,99 @@ import { Cars } from './cars.types'
...
+      where: {
+        id,
+      },
+    });
+
+    return car;
+
+  } catch (error: any){
+      throw error; 
+    }
```
Comment (ht1204) [inline src/api/cars/cars.service.ts]: balance the bracket according to code format, thx.

File: src/api/cars/cars.controller.ts [hunk 4]
```diff
@@ -1,20 +1,126 @@
...
+      luggage: carUpdated.luggage,
+      air_conditioner: carUpdated.air_conditioner,
+      transmition: carUpdated.transmition,
+      fare_km: carUpdated.fare_km,
+      driver_id: carUpdated.driver_id,
+      created_at: carUpdated.created_at,
+      updated_at: carUpdated.updated_at
+    }
+  
+    return res.json({message: "Information was updated sucessfully", data});
```
Comment (ht1204) [inline src/api/cars/cars.controller.ts]: Replace it by:
`res.status(201).json({message: "Information was updated sucessfully", data});`

File: src/api/cars/cars.controller.ts (hunk 1 above)
Comment (PVARGASM1) [reply src/api/cars/cars.controller.ts]: Ok, fixed up.

File: src/api/cars/cars.controller.ts (hunk 2 above)
Comment (PVARGASM1) [reply src/api/cars/cars.controller.ts]: Ok, fixed up.

File: src/api/cars/cars.controller.ts (hunk 4 above)
Comment (PVARGASM1) [reply src/api/cars/cars.controller.ts]: Ok, fixed up.

File: src/api/cars/cars.service.ts (hunk 3 above)
Comment (PVARGASM1) [reply src/api/cars/cars.service.ts]: changed.

**Review by ht1204 — APPROVED** [review]

**Pull request merged**